include/zephyr/net/coap_mgmt.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief CoAP Events code public header
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_COAP_MGMT_H_
+#define ZEPHYR_INCLUDE_NET_COAP_MGMT_H_
+
+#include <zephyr/net/net_mgmt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief CoAP Manager Events
+ * @defgroup coap_mgmt CoAP Manager Events
+ * @ingroup networking
+ * @{
+ */
+
+/** @cond INTERNAL_HIDDEN */
+
+/* CoAP events */
+#define _NET_COAP_LAYER		NET_MGMT_LAYER_L4
+#define _NET_COAP_CODE		0x1c0
+#define _NET_COAP_IF_BASE	(NET_MGMT_EVENT_BIT |			\
+				 NET_MGMT_LAYER(_NET_COAP_LAYER) |	\
+				 NET_MGMT_LAYER_CODE(_NET_COAP_CODE))
+
+/** @endcond */
+
+enum net_event_coap_cmd {
+	NET_EVENT_COAP_CMD_SERVICE_STARTED = 1,
+	NET_EVENT_COAP_CMD_SERVICE_STOPPED,
+};
```

Comment:
I recently added `CONFIG_COAP_OBSERVER_EVENTS` for added/removed observers, would it be a better fit to have these events added here?
Or keep them both?

Reply:
IMO it'd make sense to keep all of the CoAP-level events together. If we start to use net_mgmt framework for such events, IMO we should stick to it instead of having custom handler. So